### ISSUE TITLE
self_flow_loss computed under torch.no_grad() contributes zero gradients

### ISSUE THREAD

**Issue by CarsonBurke** [CarsonBurke/dreamer4#4]
**File(s):** `dreamer4/trainers.py:775-782`

**Details:** The entire self-flow loss computation is wrapped in `torch.no_grad()` (line 776), so the resulting tensor has `requires_grad=False`. The `SelfFlow.forward` method (in `dreamer4/dreamer4.py:2692-2714`) does not internally manage gradient context — both the teacher forward pass and the student prediction head (`self.student_predict_head`) run under the caller's `no_grad()` block. As a result, `self_flow_loss` carries no gradient, and adding it to `loss` on line 781 contributes zero gradient signal from the self-flow distillation objective. The student prediction head never learns.

```python
# trainers.py lines 775-782
if self.self_flow:
    with torch.no_grad():
        self_flow_loss = self.self_flow_module(self.ema_model.ema_model, intermediates, batch_data)

    self_flow_loss = self_flow_loss / self.grad_accum_every

    loss = loss + self_flow_loss * self.self_flow_loss_weight
    total_self_flow_loss += self_flow_loss.item()
```

**Suggested fix:** Move `torch.no_grad()` inside `SelfFlow.forward` to only wrap the teacher forward pass (the EMA model call on line 2705 of `dreamer4.py`), so the student path (`student_predict_head`) retains gradients. Alternatively, restructure the caller to only wrap the teacher call in `no_grad()`:

```python
if self.self_flow:
    self_flow_loss = self.self_flow_module(self.ema_model.ema_model, intermediates, batch_data)
    # ...
```

And inside `SelfFlow.forward`:
```python
with torch.no_grad():
    *_, teacher_intermediates = ema_teacher_model(**teacher_batch_kwargs)

teacher_hidden = teacher_intermediates.layer_hiddens[self.teacher_layer].detach()
```
